### PR TITLE
Fix compilation failure in android

### PR DESCRIPTION
**Pull request by akemimadoka**
Specify library name and version:  **lib/1.0**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by lasote** [comment]
Please, open different PRs if you want to modify several recipes

File: recipes/libpng/all/conanfile.py [hunk 1]
```diff
@@ -96,8 +96,7 @@ def package_info(self):
                 self.cpp_info.libs = ['libpng16']
         else:
             self.cpp_info.libs = ["png16"]
-            if self.settings.os == "Linux":
-                self.cpp_info.libs.append("m")
+            self.cpp_info.libs.append("m")
```
Comment (lasote) [inline recipes/libpng/all/conanfile.py]: Probably is safer to keep the "IF" adding android, otherwise other NIX systems might suffer.

File: recipes/libwebp/all/conanfile.py [hunk 2]
```diff
@@ -85,5 +85,7 @@ def package_info(self):
         self.cpp_info.libs = ['webpmux', 'webpdemux', 'webpdecoder', 'webp']
         if self.options.shared and self.settings.os == "Windows" and self.settings.compiler != 'Visual Studio':
             self.cpp_info.libs = [lib + '.dll' for lib in self.cpp_info.libs]
-        if self.settings.os == "Linux":
-            self.cpp_info.libs.append("pthread")
+        if self.settings.os != "Windows":
```
Comment (lasote) [inline recipes/libwebp/all/conanfile.py]: Same comment, limit changes to Android. It might have collateral issues for NIX systems. 
Please do different PRs to change different libraries.
Thanks!

**Comment by akemimadoka** [comment]
Thank you! I will open different PRs.
